x-pack/plugins/reporting/public/plugin.ts

```diff
@@ -124,7 +125,12 @@ export class ReportingPublicPlugin
    */
   private getApiClient(http: HttpSetup, uiSettings: IUiSettingsClient) {
     if (!this.apiClient) {
-      this.apiClient = new ReportingAPIClient(http, uiSettings, this.kibanaVersion);
+      this.apiClient = new ReportingAPIClient(
+        http,
+        uiSettings,
+        this.kibanaVersion,
+        this.config.disableStatefulSettings.enabled
```

Comment:
Could we pass the entire `config` object to the constructor instead? It would be more readable, since sometimes a parameter can be hard to understand when it's just a simple boolean.